lib/matplotlib/colors.py

```diff
@@ -364,12 +364,16 @@ def to_rgba_array(c, alpha=None):
 
     # Quick path if the whole sequence can be directly converted to a numpy
     # array in one shot.
-    lens = {len(cc) if isinstance(cc, (list, tuple)) else -1 for cc in c}
-    if lens == {3}:
-        rgba = np.column_stack([c, np.ones(len(c))])
-    elif lens == {4}:
-        rgba = np.array(c)
-    else:
+    rgba = None
+    if isinstance(c, list):
```

Comment:
maybe fast-path `tuple` as well?

Reply:
done

Reply:
Alternatively, 

```
from collections.abc import Sequence
if isinstance(c, Sequence):
    ...
```

will catch lists and tuples but not dicts, sets, or dict key/values, and I *think* it will predict whether `np.asarray` will work the way you expect.